src/librustc_codegen_llvm/common.rs

```diff
@@ -237,6 +219,13 @@ impl ConstMethods<'tcx> for CodegenCx<'ll, 'tcx> {
         unsafe { llvm::LLVMConstReal(t, val) }
     }
 
+    fn const_str(&self, s: Symbol) -> (&'ll Value, &'ll Value) {
```

Comment:
Any chance you could get rid of this by using `OperandRef::from_const` with a string literal `ty::Const`?